internal/lib/lib.go

```diff
@@ -171,6 +171,11 @@ func GetshardSize() uint32 {
 }
 
 func GetL4FqdnFormat() int32 {
+	if GetAdvancedL4() {
```

Comment:
I think this is fine for now, but we may want to support DNS for advancedL4 too based on some asks.

Reply:
It looks like this check is really moot. You could always check for the `GetAdvancedL4()` method and be done with the validation, however if you want to keep it this way - it's okay.

Reply:
this one is required to not disable autofqdn for advancedL4 since the pool node construction mechanism is common for both serviceAPI and advancedL4.